engine/common/version/version_control_test.go

```diff
@@ -0,0 +1,345 @@
+
+		// Ensure the component is ready before proceeding.
+		unittest.RequireComponentsReadyBefore(t, 2*time.Second, vc)
+
+		compatible, err := vc.CompatibleAtBlock(sealedHeader.Height)
+		require.NoError(t, err)
+
+		// Assert that the VersionControl is compatible at the specified block height.
+		assert.True(t, compatible)
+
+		compatible, err = vc.CompatibleAtBlock(startHeight - 1)
+		require.NoError(t, err)
+
+		// Should be incompatible, as lover than start height
+		assert.False(t, compatible)
+	})
+
+	t.Run("start and end heights available", func(t *testing.T) {
+		startHeight := sealedHeader.Height - 5
+
+		endHeight := sealedHeader.Height - 1
+
+		// Set up a mock version beacons storage.
+		versionBeacons := storageMock.NewVersionBeacons(t)
+		// Mock the Highest method to return a version beacon with a specific version.
+		versionBeacons.
+			On("Highest", testifyMock.AnythingOfType("uint64")).
+			Return(func(height uint64) (*flow.SealedVersionBeacon, error) {
+				if height == sealedHeader.Height {
+					return &flow.SealedVersionBeacon{
+						VersionBeacon: unittest.VersionBeaconFixture(
+							unittest.WithBoundaries(
+								flow.VersionBoundary{
+									BlockHeight: startHeight,
+									Version:     "0.0.1",
+								}),
+							unittest.WithBoundaries(
+								flow.VersionBoundary{
+									BlockHeight: endHeight,
+									Version:     "0.0.2",
+								}),
+							unittest.WithBoundaries(
+								flow.VersionBoundary{
+									BlockHeight: sealedHeader.Height,
+									Version:     "0.0.3",
+								}),
+						),
+						SealHeight: sealedHeader.Height,
+					}, nil
+				} else {
+					return nil, nil
+				}
+			}, nil)
+
+		// Create a new VersionControl instance with initial parameters.
+		vc, err := NewVersionControl(unittest.Logger(), versionBeacons, semver.New("0.0.2"), finalizedRootBlockHeight, sealedHeader.Height)
+		require.NoError(t, err)
+
+		// Create a mock signaler context for testing.
+		ictx := irrecoverable.NewMockSignalerContext(t, ctx)
+
+		// Start the VersionControl component.
+		vc.Start(ictx)
+
+		// Ensure the component is ready before proceeding.
+		unittest.RequireComponentsReadyBefore(t, 2*time.Second, vc)
+
+		compatible, err := vc.CompatibleAtBlock(sealedHeader.Height)
+		require.NoError(t, err)
+
+		// Should be incompatible, as end height is lover than sealedHeader height
+		assert.False(t, compatible)
+	})
+}
+
+// TestVersionChanged tests the behavior of the VersionControl component when the version changes.
+func TestVersionChanged(t *testing.T) {
+	ctx, cancel := context.WithCancel(context.Background())
+	defer cancel()
+
+	sealedHeader := unittest.BlockHeaderFixture(unittest.WithHeaderHeight(10))
+	sealedVersionBeacon := &flow.SealedVersionBeacon{
+		VersionBeacon: unittest.VersionBeaconFixture(
+			unittest.WithBoundaries(
+				flow.VersionBoundary{
+					BlockHeight: sealedHeader.Height,
+					Version:     "0.0.1",
+				}),
+		),
+		SealHeight: sealedHeader.Height,
+	}
```

Comment:
suggestion:
Create a new method:
```golang
func versionBeaconEvent(height uint64, version string) *flow.SealedVersionBeacon {
	return &flow.SealedVersionBeacon{
		VersionBeacon: unittest.VersionBeaconFixture(
			unittest.WithBoundaries(
				flow.VersionBoundary{
					BlockHeight: height,
					Version:     version,
				}),
		),
		SealHeight: height,
	}
}
```

then you can create new events with minimal boiler plate:
```golang
sealedVersionBeacon := versionBeaconEvent(sealedHeader.Height, "0.0.1")
```